packages/core/src/2d/sprite/Sprite.ts

```diff
@@ -44,21 +46,27 @@ export class Sprite extends RefObject {
     if (this._texture !== value) {
       this._texture = value;
       this._dispatchSpriteChange(SpriteModifyFlags.texture);
-      (this._width === undefined || this._height === undefined) && this._dispatchSpriteChange(SpriteModifyFlags.size);
+      if (this._customWidth === undefined || this._customHeight === undefined) {
+        this._dispatchSpriteChange(SpriteModifyFlags.size);
+      }
     }
   }
 
   /**
    * The width of the sprite (in world coordinates).
    */
   get width(): number {
-    this._width === undefined && this._calDefaultSize();
-    return this._width;
+    if (this._customWidth !== undefined) {
```

Comment:
According to the current trend, we can no longer use unspoken rules:
1. First of all, you need to write clearly what the "trigger conditions" of the two modes are
2. What is the difference between the two